src/DependencyInjection/Configuration.php

```diff
@@ -18,8 +19,14 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('m6_statsd');
+        // >= 4.2.0
+        if ((int) Kernel::VERSION_ID >= 40200) {
+            $treeBuilder = new TreeBuilder('m6_statsd');
+            $rootNode = $treeBuilder->getRootNode();
+        } else {
+            $treeBuilder = new TreeBuilder();
+            $rootNode = $treeBuilder->root('m6_statsd');
+        }
```

Comment:
please check for `getRootNode` method existence instead

```suggestion
        // symfony/config >= 4.2.0
        if (\method_exists($treeBuilder, 'getRootNode')) {
            $treeBuilder = new TreeBuilder('m6_statsd');
            $rootNode = $treeBuilder->getRootNode();
        } else {
            $treeBuilder = new TreeBuilder();
            $rootNode = $treeBuilder->root('m6_statsd');
        }
```